CMake/VeloxUtils.cmake

```diff
@@ -93,13 +93,26 @@ function(velox_link_libraries TARGET)
   # TODO(assignUser): Handle scope keywords (they currently are empty calls ala
   # target_link_libraries(target PRIVATE))
   if(VELOX_MONO_LIBRARY)
-    message(DEBUG "${TARGET}: ${ARGN}")
-    foreach(_lib ${ARGN})
-      if("${_lib}" MATCHES "^velox_*")
-        message(DEBUG "\t\tDROP: ${_lib}")
+    # These targets follow the velox_* name for consistency but are NOT actually
+    # aliases to velox when building the mono lib and need to be linked
+    # explicitly (this is a hack)
+    set(explicit_targets
+        velox_exec_test_lib
+        # see velox/experimental/wave/README.md
+        velox_wave_common
+        velox_wave_decode
+        velox_wave_dwio
+        velox_wave_exec
+        velox_wave_stream
+        velox_wave_vector)
+
+    foreach(_arg ${ARGN})
+      list(FIND explicit_targets ${_arg} _explicit)
+      if(_explicit EQUAL -1 AND "${_arg}" MATCHES "^velox_*")
+        message(DEBUG "\t\tDROP: ${_arg}")
       else()
-        message(DEBUG "\t\tADDING: ${_lib}")
-        target_link_libraries(velox ${_lib})
+        message(DEBUG "\t\tADDING: ${_arg}")
```

Comment:
That's right. We normally drop all `velox_*` targets because they are all just an `ALIAS` to `velox` but some of the targets like exec_test_lib and the wave (to avoid cuda driver issues) libs are not and need to be linked.

I didn't want us to make the naming conventions inconsistent, therefor this hack.